### ISSUE TITLE
Dealer POD data needed for Dealer Locator

### ISSUE THREAD

**Issue by BeckyMedlin** [hlxsites/vg-volvotrucks-us#467]
Dealer Locator functionality needs to consume data from the Dealer POD. 
Dealer POD API contact: Ken Johnson

**Comment by KJohnsn** [comment]
**Description**

The data source for the dealer locator will need to be replaced. Currently, it uses an MQ Feed to provide the information which is displayed when a user indicates what their location is and who the nearest dealers are. This helps to inform the customer who they can contact for specific questions regarding their current need.

The MQ Feed is going to be removed from the dealer locator and a Dealer POD will be used instead. 
POD stands for Portable Operational Data and the Dealer POD API would need to be requested and then implemented to ensure that the new dealer data is consumed and displayed accordingly when requested.

The dealer data will be for Volvo Trucks

**Actions**
•	Request the Dealer POD needed - Ken
•	Provide Dealer POD API - Ken
•	Replace MQ feed with Dealer POD API to dealer locator - Netcentric

**Acceptance Criteria**
•	Dealer information displays correctly in the dealer locator
•	Dealer information includes North American countries with active class codes “US”, “CDN”, and “MEX”
•	Dealers are indicated by specific color markers based on their category: Dealer, Certified Uptime Dealer, Certified Electric Dealer
•	Dealer indicators on the map need to be shown clearly and positioned on the correct location
•	Dealer information is provided when a person clicks on More
•	Dealer information is clear and easy to read
•	The title for the dealer is clearly displayed
•	The service icons are clear
•	The dealer location including opening or closing hour is displayed (12h format)
•	Services are displayed
•	Driver Amenities is displayed

**Comment by Gonzalo-CG** [comment]
Since the POD API is still unavailable, this issue was estimated at 2 SP assuming it implies only a change in configuration and testing (the new API has the same payload). If the new POD API leads to other significant modifications, this issue will need to be re-estimated and probably moved to the next sprint.

**Comment by KJohnsn** [comment]
Discussion on complexity of the Dealer POD development needed for this is underway.

**Comment by KJohnsn** [comment]
@Joesturbo Dealer locator address for each brand to mask the URL properly. Then once this is done, then we will share the URL in a new comment.

**Comment by KJohnsn** [comment]
The JSON for Volvo is: https://as-dealerloc-endpoint-prod-001.azurewebsites.net/Volvo_DealerJSON.ashx and the name for this URL would be DL.volvotrucks.us. Needs to point to this address. This is a test URL so we can validate that the data set is right.

**Comment by SantiagoHomps-NC** [comment]
In this case we are missing 31 names, 170 phones and 397 addresses.
Some examples attached.
<img width="572" alt="Screenshot 2024-04-11 at 16 20 23" src="https://github.com/hlxsites/vg-volvotrucks-us/assets/103571932/ad01738b-e02c-4c84-a7f6-b169d037a0b1">
[failing-dealers-volvo.js.zip](https://github.com/hlxsites/vg-volvotrucks-us/files/14947546/failing-dealers-volvo.js.zip)

**Comment by KJohnsn** [comment]
@SantiagoHomps-NC try https://as-dealerloc-endpoint-prod-001.azurewebsites.net/Volvo_DealerJSON.ashx

**Comment by verareljic** [comment]
Will be taken up the next 5 weeks according to @Kannan-CM

**Comment by KJohnsn** [comment]
@SantiagoHomps-NC not sure if you saw my comment above but try this updated version which should show the dealer name - https://as-dealerloc-endpoint-prod-001.azurewebsites.net/Volvo_DealerJSON.ashx

**Comment by SantiagoHomps-NC** [comment]
@Kannan-CM 
Same CORS issue here 
<img width="384" alt="Screenshot 2024-04-19 at 14 41 37" src="https://github.com/hlxsites/vg-volvotrucks-us/assets/103571932/385f8c1b-8643-4b83-a6aa-b2f625209c1c">

**Comment by Kannan-CM** [comment]
Got an update stating the SSL cert did not work.
The live URL should be ready by Monday, this time

**Comment by Kannan-CM** [comment]
@SantiagoHomps-NC ,

SSL applied for:
https://dealerlocator.volvotrucks.us/Volvo_DealerJson.ashx

**Comment by SantiagoHomps-NC** [comment]
@Kannan-CM I still get the CORS error here 
<img width="656" alt="Screenshot 2024-04-25 at 12 42 28" src="https://github.com/hlxsites/vg-volvotrucks-us/assets/103571932/a4f6101d-7668-4a31-adb2-ccb02d5c83af">

**Comment by Kannan-CM** [comment]
Thanks for Falgging @SantiagoHomps-NC ,
This should be resolved now.

**Comment by girishdigrajkar** [comment]
CORS error is not displayed anymore.

**Outstanding Issues:**
1. Complete address is not displayed:
![image](https://github.com/hlxsites/vg-volvotrucks-us/assets/125972657/0ee99be3-ef1e-48e3-b17c-fdecfa7f0287)

2. The address should be on multiple lines as implementation on live site 
![image](https://github.com/hlxsites/vg-volvotrucks-us/assets/125972657/12458f2e-ab30-469e-94ee-cc6dcb37bc54)

5. Rental & Leasing information is not displayed:
      Navigate to https://467-dealer-pod--vg-volvotrucks-us--hlxsites.hlx.page/drafts/shomps/dealer
      Search for 'New york" > list of dealer will be displayed; click on filter icon, select filter by service "Rental & Leasing" click 
     on back and validate:
     List is displayed empty
![image](https://github.com/hlxsites/vg-volvotrucks-us/assets/125972657/16dda940-72d0-448a-9780-c05b9789d35e)

**Comment by santi-homps** [comment]
@girishdigrajkar after a lot of investigation I found that there are no Leasing dealers in this new API. That's why we are not displaying none. Is this ok @Kannan-CM or should we take a closer look? in case no leasing dealers are present, would it make sense to even have this filtering option?
Below a screenshot of all the DEALER_TYPE_DESC listed:
<img width="859" alt="Screenshot 2024-05-02 at 18 23 35" src="https://github.com/hlxsites/vg-volvotrucks-us/assets/68013336/ae0c3d94-eefb-4b03-af08-ff14f06ecfe4">

**Comment by santi-homps** [comment]
also address issue has been solved @girishdigrajkar

**Comment by mollyaljenkins** [comment]
This ticket is on hold and needs to be discussed between Scott, @girishdigrajkar @SantiagoHomps-NC

**Comment by BeckyMedlin** [comment]
@santi-homps For your comment regarding the address issue -- does this mean items 1 and 2 noted above with screenshots are no longer a problem? @girishdigrajkar

**Comment by KJohnsn** [comment]
@santi-homps For launch: Due to lack of export ability out of NACDB to Data pods for leasing data, we will remove the filter functionality to simplify the user experience **temporarily**.
We will keep default radius at 75 miles
Also noticed that one of the dealers in Mexico have the same contact number for multiple locations. @Kannan-CM and @KJohnsn are speaking with Catelyn to investigate why this is.

**Comment by santi-homps** [comment]
Added fix for hover state in share button
https://github.com/hlxsites/vg-volvotrucks-us/assets/68013336/ddae4b80-76c2-48ae-8dc3-77cf781f3c65

**Comment by KJohnsn** [comment]
@santi-homps sharing the URL that was placed in the Teams Chat https://dealerlocator.volvotrucks.us/Volvo_DealerJson.ashx
